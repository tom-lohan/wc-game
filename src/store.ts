import { configureStore } from "@reduxjs/toolkit";
import gameReducer from './wcgSlice';

export const store = configureStore({
	reducer: {
		wcgame: gameReducer
	},
});


export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;