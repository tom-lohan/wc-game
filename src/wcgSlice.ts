import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { RootState } from "./store";
import { Country, Player } from "./types";
import { countries } from './data/counties';

export interface WCGState {
	players: Player[],
	countries: Country[],
	includedCountries: Country[],
	excludedCountries: Country[]
}

export const WCGInitialState: WCGState = {
	players: [],
	countries: countries,
	includedCountries: [],
	excludedCountries: []
}

export const WCGSlice = createSlice({
	name: 'wcgame',
	initialState: WCGInitialState,
	reducers: {
		addPlayer: (state, action: PayloadAction<Player>) => {
			return {
				...state,
				players: state.players.concat([action.payload])
			}
		},
		removePlayerByName: (state, action: PayloadAction<string>) => {
			return {
				...state,
				players: state.players.filter(player => player.name !== action.payload)
			}
		},
		setPlayers: (state, action: PayloadAction<Player[]>) => {
			return {
				...state,
				players: action.payload
			}
		},
		setIncludedCountries: (state, action: PayloadAction<Country[]>) => {
			return {
				...state,
				includedCountries: action.payload
			}
		},
		setExcludedCountries: (state, action: PayloadAction<Country[]>) => {
			return {
				...state,
				excludedCountries: action.payload
			}
		}
	}
});


export const selectCurrentPlayers = (state: RootState) => state.wcgame.players;
export const selectCurrentCountries = (state: RootState) => state.wcgame.countries;
export const selectRankedCountries = (state: RootState) => [...state.wcgame.countries].sort((cA, cB) => cA.ranking < cB.ranking ? -1 : 1);
export const selectIncludedCountries = (state: RootState) => state.wcgame.includedCountries;
export const selectExcludedCountries = (state: RootState) => state.wcgame.excludedCountries;

export const { addPlayer, removePlayerByName, setPlayers, setExcludedCountries, setIncludedCountries } = WCGSlice.actions;

export default WCGSlice.reducer;