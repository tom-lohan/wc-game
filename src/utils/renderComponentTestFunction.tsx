import { configureStore } from '@reduxjs/toolkit'
import { render } from '@testing-library/react'
import React, { ReactNode } from 'react'
import { Provider } from 'react-redux'
import '@testing-library/jest-dom'
import { WCGInitialState } from '../wcgSlice'
import gameReducer from '../wcgSlice'

export const renderComponentTestFunction = (
    children: ReactNode,
    modifiedStore: object = {}
) => {
    const store = {
        wcgame: { ...WCGInitialState },
    }

    const mergedStore = { ...store, ...modifiedStore }

    const mockedStore = configureStore({
        reducer: {
            wcgame: gameReducer,
        },
        preloadedState: mergedStore,
        middleware: (getDefaultMiddleware) =>
            getDefaultMiddleware({ serializableCheck: false }),
    })

    return render(<Provider store={mockedStore}>{children}</Provider>)
}
