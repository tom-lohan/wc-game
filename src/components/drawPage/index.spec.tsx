import { DrawPage } from '.'
import { RootState } from '../../store'
import { renderComponentTestFunction } from '../../utils/renderComponentTestFunction'
import { WCGInitialState } from '../../wcgSlice'

const mockedSetSelectedPage = jest.fn()

describe('DrawPage', () => {
    // let modifiedStore: RootState
    beforeEach(() => {
        // modifiedStore = {
        //     wcgame: {
        //         ...WCGInitialState,
        //     },
        // }
        jest.clearAllMocks()
        jest.restoreAllMocks()
    })

    it('should render correctly', () => {
        const { container } = renderComponentTestFunction(
            <DrawPage setSelectedPage={mockedSetSelectedPage} />
        )
        expect(container).toMatchSnapshot()
    })
})
