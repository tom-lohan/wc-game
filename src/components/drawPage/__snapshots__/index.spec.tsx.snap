// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawPage should render correctly 1`] = `
<div>
  <main
    class="drawPage p-5"
  >
    <button
      class="ant-btn ant-btn-default"
      type="button"
    >
      <span>
        Back
      </span>
    </button>
    <div
      class="flex gap-7 mt-5"
    >
      <div
        class="players "
      >
        <h2
          class="text-3xl"
        >
          Players
        </h2>
        <div
          class="playerList flex gap-4 flex-wrap"
        />
      </div>
      <div
        class="countries"
      >
        <div
          class="flex gap-3 items-baseline"
        >
          <h2
            class="text-3xl"
          >
            Countries
          </h2>
          <span>
            Ordered using
             
            <a
              href="https://www.theguardian.com/football/2022/jun/15/world-cup-2022-power-rankings-final-32-qatar"
              rel="noreferrer"
              target="_blank"
            >
              the Guardian's power rankings
            </a>
             
          </span>
        </div>
        <div
          class="countryList flex gap-4 flex-wrap"
        >
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Argentina
              </span>
              <span
                class="text-xl"
              >
                🇦🇷
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Brazil
              </span>
              <span
                class="text-xl"
              >
                🇧🇷
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Spain
              </span>
              <span
                class="text-xl"
              >
                🏴󠁧󠁢󠁥󠁮󠁧󠁿
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Netherlands
              </span>
              <span
                class="text-xl"
              >
                🇳🇱
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Germany
              </span>
              <span
                class="text-xl"
              >
                🇩🇪
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                France
              </span>
              <span
                class="text-xl"
              >
                🇫🇷
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Denmark
              </span>
              <span
                class="text-xl"
              >
                🇩🇰
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Belgium
              </span>
              <span
                class="text-xl"
              >
                🇧🇪
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Portugal
              </span>
              <span
                class="text-xl"
              >
                🇵🇹
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                England
              </span>
              <span
                class="text-xl"
              >
                🏴󠁧󠁢󠁥󠁮󠁧󠁿
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Senegal
              </span>
              <span
                class="text-xl"
              >
                🇸🇳
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Uruguay
              </span>
              <span
                class="text-xl"
              >
                🇺🇾
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Croatia
              </span>
              <span
                class="text-xl"
              >
                🇭🇷
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Switzerland
              </span>
              <span
                class="text-xl"
              >
                🇨🇭
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Serbia
              </span>
              <span
                class="text-xl"
              >
                🇷🇸
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Canada
              </span>
              <span
                class="text-xl"
              >
                🇨🇦
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Wales
              </span>
              <span
                class="text-xl"
              >
                🏴󠁧󠁢󠁷󠁬󠁳󠁿
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Poland
              </span>
              <span
                class="text-xl"
              >
                🇵🇱
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Mexico
              </span>
              <span
                class="text-xl"
              >
                🇲🇽
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                USA
              </span>
              <span
                class="text-xl"
              >
                🇺🇸
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Morocco
              </span>
              <span
                class="text-xl"
              >
                🇲🇦
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Cameroon
              </span>
              <span
                class="text-xl"
              >
                🇨🇲
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Iran
              </span>
              <span
                class="text-xl"
              >
                🇮🇷
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Tunisia
              </span>
              <span
                class="text-xl"
              >
                🏴󠁧󠁢󠁥󠁮󠁧󠁿
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Japan
              </span>
              <span
                class="text-xl"
              >
                🇯🇵
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Ecuador
              </span>
              <span
                class="text-xl"
              >
                🇪🇨󠁧󠁢󠁥
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Ghana
              </span>
              <span
                class="text-xl"
              >
                🇬🇭
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                South Korea
              </span>
              <span
                class="text-xl"
              >
                🇰🇷
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Costa Rica
              </span>
              <span
                class="text-xl"
              >
                🇨🇷
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Qatar
              </span>
              <span
                class="text-xl"
              >
                🇶🇦
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Australia
              </span>
              <span
                class="text-xl"
              >
                🇦🇺
              </span>
            </div>
          </div>
          <div
            class="playerCard flex items-center justify-center border-2 border-black p-3"
          >
            <div
              class="m-0 flex gap-2 justify-center items-center"
            >
              <span>
                Saudi Arabia
              </span>
              <span
                class="text-xl"
              >
                🇸🇦
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
